File: testcases/benchmark/searching/linear_search.cpp
#include <iostream>
#include <vector>

int linear_search(const std::vector<int>& arr, int target) {
    for (int i = 0; i < (int)arr.size(); i++) {
        if (arr[i] == target) return i;
    }
    return -1;
}

int main() { return 0; }

File: testcases/benchmark/searching/linear_search.f90
function linear_search(arr, n, target) result(idx)
    integer, intent(in) :: arr(:), n, target
    integer :: idx, i
    idx = -1
    do i = 1, n
        if (arr(i) == target) then
            idx = i
            return
        end if
    end do
end function linear_search

program main
end program main
